### DISCUSSION TITLE
Currency issues

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#16806] [Feature improvement]
I have noticed that determining currency via user settings bringing sometimes issues. For instance:  invoice  has been created with different currency than it is set for user. This invoice has also custom currency fields calculated on the basis of sum_gross (for example).
In this case custom fields are displayed in user currency instead of currency specified for invoice line items.
This bringing confusion in PDF or Email Template outputs.
What I would propose is to create new UI currency type modeled as follows:
DECIMAL FIELD + CURRENCY SELECTION FIELD
The same way as Salutation is deployed in Contacts module, so that currency can be specified in-field

Unless here is some alternative method I didn't recognize yet?

**Comment by s2b-git** [comment]
It also impacts system fields displaying way as per screenshot:

![image](https://user-images.githubusercontent.com/84110093/212217471-e39ad698-f014-4c68-94a6-4ca1b2e5dcf0.png)

**Reply by mariuszkrzaczkowski** [reply]
there is some workaround, add a field in this module called currency_id
Same as in the deposit module
https://gitdeveloper.yetiforce.com/index.php?module=LayoutEditor&parent=Settings&view=Index&sourceModule=PaymentsIn

**Comment by newminder** [comment]
did you solve this?
